Zend/zend_execute_API.c

```diff
@@ -253,136 +246,124 @@ void shutdown_destructors(void) /* {{{ */
 
 void shutdown_executor(void) /* {{{ */
 {
-	zend_function *func;
-	zend_class_entry *ce;
+	zend_string *key;
+	zval *zv;
+#if ZEND_DEBUG
+	zend_bool fast_shutdown = 0;
+#else
+	zend_bool fast_shutdown = is_zend_mm() && !EG(full_tables_cleanup);
+#endif
 
 	zend_try {
+		zend_llist_destroy(&CG(open_files));
```

Comment:
Can we move the `EG(active) = 0` to the top of this function instead of the bottom? We had issues in the past where objects could be created after destructors were already run. Setting active=0 early should make completely sure that no user code will run.